src/main/java/data_access/DataGateway.java

```diff
@@ -45,11 +47,11 @@ public static List getArtistIds(JsonNode data) {
     }
 
     // returns the genre of all artists of the tracks in a user's top tracks
-    public static List<String> getArtistGenre(List<String> trackIdsList) throws IOException {
+    public static List<String> getArtistGenre(List<String> trackIdsList, CurrentUser currentUser) throws IOException {
         String artist_url = "https://api.spotify.com/v1/artists/";
         List<String> genresList = new ArrayList<>();
         for (int i = 0; i < trackIdsList.size(); i++) {
-            JsonNode data = DataGetterClass.getData(artist_url + trackIdsList.get(i));
+            JsonNode data = DataGetterClass.getData(artist_url + trackIdsList.get(i), currentUser);
             for (int j = 0; j < data.get("genres").size(); j++) {
                 if (data.get("genres").get(j) != null) {
                     genresList.add(data.get("genres").get(j).asText()); // Convert JsonNode to String
```

Comment:
Since we were using the DataGetter classes to send the requests and the interactor class to implement the functionality, is the Data Gateway class needed or can this functionality be implemented in another existing class (such as the Implementor class) to fit the general structure of the project?